File: brasil_burger/src/main/java/sn/ism/brasilburger/entity/Menu.java
package sn.ism.brasilburger.entity;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

public class Menu {
    private int id;
    private String nom;
    private String image;
    private boolean archive;
    private LocalDateTime dateCreation;
    private LocalDateTime dateModification;
    private List<CompositionMenu> compositions;
    private double prixTotal;

    // Constructeurs
    public Menu() {
        this.compositions = new ArrayList<>();
    }

    public Menu(String nom, String image) {
        this.nom = nom;
        this.image = image;
        this.archive = false;
        this.compositions = new ArrayList<>();
    }

    // Getters et Setters
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getNom() { return nom; }
    public void setNom(String nom) { this.nom = nom; }

    public String getImage() { return image; }
    public void setImage(String image) { this.image = image; }

    public boolean isArchive() { return archive; }
    public void setArchive(boolean archive) { this.archive = archive; }

    public LocalDateTime getDateCreation() { return dateCreation; }
    public void setDateCreation(LocalDateTime dateCreation) { this.dateCreation = dateCreation; }

    public LocalDateTime getDateModification() { return dateModification; }
    public void setDateModification(LocalDateTime dateModification) { this.dateModification = dateModification; }

    public List<CompositionMenu> getCompositions() { return compositions; }
    public void setCompositions(List<CompositionMenu> compositions) { this.compositions = compositions; }

    public double getPrixTotal() { return prixTotal; }
    public void setPrixTotal(double prixTotal) { this.prixTotal = prixTotal; }

    public void ajouterComposition(CompositionMenu composition) {
        this.compositions.add(composition);
    }

    @Override
    public String toString() {
        return String.format("Menu{id=%d, nom='%s', prixTotal=%.2f FCFA, archive=%s}",
                id, nom, prixTotal, archive ? "Oui" : "Non");
    }

}

File: src/Entity/Menu.php
<?php
namespace App\Entity;

use App\Repository\MenuRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: MenuRepository::class)]
#[ORM\Table(name: 'menu')]
#[ORM\Index(name: 'idx_menu_archive', columns: ['archive'])]
#[ORM\HasLifecycleCallbacks]
class Menu
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: 'id_menu')]
    private ?int $idMenu = null;

    #[ORM\Column(length: 100)]
    private ?string $nom = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $image = null;

    #[ORM\Column(type: 'boolean')]
    private ?bool $archive = false;

    #[ORM\Column(name: 'date_creation', type: 'datetime')]
    private ?\DateTimeInterface $dateCreation = null;

    #[ORM\Column(name: 'date_modification', type: 'datetime')]
    private ?\DateTimeInterface $dateModification = null;

    #[ORM\OneToMany(mappedBy: 'menu', targetEntity: CompositionMenu::class, cascade: ['persist', 'remove'])]
    private Collection $compositions;

    #[ORM\OneToMany(mappedBy: 'menu', targetEntity: LigneCommande::class)]
    private Collection $lignesCommande;

    public function __construct()
    {
        $this->dateCreation = new \DateTime();
        $this->dateModification = new \DateTime();
        $this->archive = false;
        $this->compositions = new ArrayCollection();
        $this->lignesCommande = new ArrayCollection();
    }

    #[ORM\PreUpdate]
    public function preUpdate(): void
    {
        $this->dateModification = new \DateTime();
    }

    public function getIdMenu(): ?int
    {
        return $this->idMenu;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;
        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(?string $image): self
    {
        $this->image = $image;
        return $this;
    }

    public function getArchive(): ?bool
    {
        return $this->archive;
    }

    public function setArchive(bool $archive): self
    {
        $this->archive = $archive;
        return $this;
    }

    public function getDateCreation(): ?\DateTimeInterface
    {
        return $this->dateCreation;
    }

    public function setDateCreation(\DateTimeInterface $dateCreation): self
    {
        $this->dateCreation = $dateCreation;
        return $this;
    }

    public function getDateModification(): ?\DateTimeInterface
    {
        return $this->dateModification;
    }

    public function setDateModification(\DateTimeInterface $dateModification): self
    {
        $this->dateModification = $dateModification;
        return $this;
    }

    public function getCompositions(): Collection
    {
        return $this->compositions;
    }

    public function addComposition(CompositionMenu $composition): self
    {
        if (!$this->compositions->contains($composition)) {
            $this->compositions[] = $composition;
        }
        return $this;
    }

    public function getLignesCommande(): Collection
    {
        return $this->lignesCommande;
    }
}